Sources/TuistKit/Services/GraphService.swift

```diff
@@ -98,3 +109,39 @@ final class GraphService {
         try System.shared.runAndPrint(["brew", "install", "graphviz"], verbose: false, environment: env)
     }
 }
+
+private enum GraphServiceError: FatalError {
+    case bug(String)
+    case encodingError(String)
+    
+    var description: String {
+        switch self {
+        case .bug(let message):
+            return message
+        case .encodingError(let format):
+            return "failed to encode graph to \(format)"
+        }
+    }
+    
+    var type: ErrorType {
+        switch self {
+        case .bug:
+            return .abort
+        case .encodingError:
+            return .abort
+        }
+    }
+}
+
+fileprivate extension GraphOutput {
+
+    func export(to filePath: AbsolutePath) throws {
+        let jsonData = try JSONEncoder().encode(self)
```

Comment:
Using `.sortedKeys` will make the output consistent, allowing for string comparison.

I also suggest using `.prettyPrinted` to match your sample in the docs and ` .withoutEscapingSlashes` to remove the unnecessary slashes.

```suggestion
        let encoder = JSONEncoder()
        encoder.outputFormatting = [.sortedKeys, .prettyPrinted, .withoutEscapingSlashes]
        let jsonData = try encoder.encode(self)
```

Reply:
Sounds good!